add resources page

MrJithil opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/nodejs/nodejs.dev/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/nodejs/nodejs.dev/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Description
Create a new /resources page with the same content as https://github.com/nodejs/nodejs.org/edit/master/locale/en/about/resources.md 
<!-- Write a brief description of the changes introduced by this PR -->

![image](https://user-images.githubusercontent.com/26359740/119044520-ee2d8580-b9d7-11eb-9aa2-dc2097d1b009.png)

URL to the added page : https://staging.nodejs.dev/1347/resources/
## Related Issues
#1341 
<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234, Addresses #1234, Related to #1234, etc.
-->

<!--
  If you want to generate a preview of this PR on our staging server please
  make a comment on the Pull-Request with the text `/preview`
 -->

MrJithil commented:
@benhalverson help needed here to map the SVG.

MrJithil commented:
> @benhalverson help needed here to map the SVG.

Issue resolved.

MrJithil commented:
Need to regenerate the preview after broken link fixes

manishprivet reviewed (commented):
In porting the about section of nodejs.org, we've been putting the related .md files in the `/content/about` folder. I think it would be better to keep the `resources.md` file also there.

MrJithil commented:
> In porting the about section of nodejs.org, we've been putting the related .md files in the `/content/about` folder. I think it would be better to keep the `resources.md` file also there.

done

benhalverson commented on content/about/resources.md:
```diff
@@ -0,0 +1,32 @@
+Please review the [trademark policy](/trademark) for information about permissible use of Node.js® logos and marks.
+
+Guidelines for the visual display of the Node.js mark are described in
+the [Visual Guidelines](/files/foundation-visual-guidelines.pdf).
```
This link is broken.

MrJithil replied:
It's already amended. We may need to regenerate the preview.

MrJithil replied:
Will look into it after a new preview generation.

benhalverson commented on src/styles/layout.scss:
```diff
@@ -530,3 +530,21 @@ details {
+}
+
+.bg-white {
+  background-color: #fff !important;
```
Can you be more specific and not use `!important`? If not add a comment on why

MrJithil replied:
Removed that. I was using the same style as the nodejs src. Removed that,

benhalverson reviewed (approved):
👍🏻 LGTM

joesepi commented:
Needs left sidebar.

joesepi commented:
Is there a reason we are putting files (pdf) and images (ai) in these folders (files/images) in an about directory? In the current site (nodejs.org) and in most sites that I've built, we have a top level directory for static files and things are organized in there. That is how the current site is. I'm confused about putting files in with the pages. Perhaps that is a Gatsby/React thing?

https://github.com/nodejs/nodejs.org/tree/master/static

MrJithil commented:
> Is there a reason we are putting files (pdf) and images (ai) in these folders (files/images) in an about directory? In the current site (nodejs.org) and in most sites that I've built, we have a top level directory for static files and things are organized in there. That is how the current site is. I'm confused about putting files in with the pages. Perhaps that is a Gatsby/React thing?
> 
> 
> 
> https://github.com/nodejs/nodejs.org/tree/master/static

Yeah. You are right. The gatsby doesn't alow us to directly link the files out of the markdown file location.

To use the files outside the content folder, we need to rewrite the html part to React TSX files. 

I guess, its better to be with the markdowns rather than create a TSX to carry the contents.

benhalverson commented:
if we add 
```
  `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          {
            resolve: `gatsby-remark-images`,
          },
        ],
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/images`,
      },
    },
``` 
to gatsby-config it should allow us to reference the images from our images folder in the src folder.
 Here's some documentation about referencing images in markdown with Gatsby. https://www.gatsbyjs.com/docs/working-with-images-in-markdown/#using-the-transformer-remark-plugin

MrJithil commented:
> if we add
> 
> ```
>   `gatsby-plugin-sharp`,
>     {
>       resolve: `gatsby-transformer-remark`,
>       options: {
>         plugins: [
>           {
>             resolve: `gatsby-remark-images`,
>           },
>         ],
>       },
>     },
>     {
>       resolve: `gatsby-source-filesystem`,
>       options: {
>         path: `${__dirname}/src/images`,
>       },
>     },
> ```
> 
> to gatsby-config it should allow us to reference the images from our images folder in the src folder.
> Here's some documentation about referencing images in markdown with Gatsby. https://www.gatsbyjs.com/docs/working-with-images-in-markdown/#using-the-transformer-remark-plugin

These configs already exist in our repo. But, while trying that, gatsby not copying the files to the public. BTW, will try it one more time Tomorrow.

MrJithil commented:
I have added the gatsby config to map the images and files from the external path. It's working. BTW, I would like to get suggestions here.

1. I have added the downloadable logos to the path `images/logos/downloads/*` - here, the extra folder `downloads` created for better housekeeping. Since the 8 files are so identical in names and in the future, if we need to replace them, it will be confused if it's placed in the `logos` folder. In another case, we don't want to copy all the contents of the logo folder to `public`. Please suggest if any better path or place for these files.
2. Placed the visual guidelines pdf file in the `resources` of the `docs` folder. Here, also, we don't want to copy all files to the public. Instead, we can only expose the files inside the resources folder. Please comment or suggest any better location for these files.

CC: @joesepi @benhalverson

joesepi commented:
Did we discuss that the concept of `static` is a thing in Gatsby, so to mirror current Nodejs.org site, we would move this PDF to `static/static/resources`? cc @benhalverson

MrJithil commented:
> Did we discuss that the concept of `static` is a thing in Gatsby, so to mirror current Nodejs.org site, we would move this PDF to `static/static/resources`? cc @benhalverson

Sounds good.

In my opinion, instead of `static/static/resources`, can we go ahead with `static/static/documents`? It will be more identical to the NodeJs.org URL.

MrJithil commented:
A bunch of optimizations added. Please create /preview before reviewing the new page.
